packages/amplify-graphql-index-transformer/src/resolvers.ts

```diff
@@ -541,7 +542,8 @@ function makeQueryResolver(config: IndexDirectiveConfiguration, ctx: Transformer
       `${queryTypeName}.${queryField}.{slotName}.{slotIndex}.res.vtl`,
     ),
   );
-  resolver.mapToStack(table.stack);
+
+  resolver.mapToStack(ctx.stackManager.getStackFor(resolverResourceId, table.stack.node.id));
```

Comment:
Coming from https://github.com/aws-amplify/amplify-category-api/pull/1083/files#r1053586297 .

Changed `table.stack.node._actualNode.id` to `table.stack.node.id` as `_actualNode` is not present in new CDK/construct lib.

Tested locally from sources with test added in referenced PR.
<img width="1113" alt="image" src="https://user-images.githubusercontent.com/5849952/208741993-958ac571-e3fc-4d73-b111-3fb89bf53b4a.png">